images/ubuntu-tools/Dockerfile

```
@@ -1,52 +1,30 @@
-        && chmod 0440 "/etc/sudoers.d/${USERNAME}" \
-        && visudo -cf "/etc/sudoers.d/${USERNAME}"; \
+# 2. Consolidated User Creation & Permissions
+RUN if [ "${USERNAME}" != "root" ] && ! id -u ${USERNAME} > /dev/null 2>&1; then \
+        useradd -m -s /bin/zsh ${USERNAME} \
+        && usermod -aG sudo ${USERNAME} \
+        && echo "${USERNAME} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers; \
     fi
 
-# 2. Install Packages, Antigravity CLI, and Clean Up (with BuildKit Caching)
+# 3. OS Packages & Antigravity CLI (BuildKit Cached)
 RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
```

Hadolint (2.14.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)